### ISSUE TITLE
Suggested improvements

### ISSUE THREAD

**Issue by BobRay** [tasianmedia/cookieJar#11]
I included a mention of your extra in a blog post, and after looking at the code, had a few ideas for you I wanted to mention:

1.  Make the expiration in days rather than seconds by multiplying the value 86400. Users can still use fractional days if they want a shorter period.

`$expires = !empty($expires) ? time() + intval( ($expires * 86400)) : '';`

2. Add the cookie name and value to the $_COOKIE array so the value will be available later in the page without a reload.

3. Log an error if setCookie() fails.

```
if (isset($name)) {
  $success = setcookie($name,$value,intval($expires),$path,$domain,$secure,$httponly);
} else {
  $modx->log(modX::LOG_LEVEL_ERROR, 'setCookie() - &name is required');
}

if ($success) {
    /* Make cookie available immediately */
   $_COOKIE[$name] = $value;
} else {
   $modx->log(modX::LOG_LEVEL_ERROR, 'Cookie not set');    
}
```
I didn't do a pull request because I wasn't sure what you'd want to do about existing installs that use seconds (e.g., convert them in the update package, or introduce a new property: &units which defaults to seconds).

**Comment by davidpede** [comment]
Thanks for these suggestions @BobRay and the mention! Will definitely look to include in next release.

Do you have a link to your article?

**Comment by BobRay** [comment]
The article won't be published until some time in December, so I don't have a link to it yet. The current title is Passing Data between Two Snippets IV.

**Comment by davidpede** [comment]
Hi @BobRay take a look at PR #14, think it covers the improvements brought up above. Any feedback welcome - looking to release in next couple of days.

**Comment by davidpede** [comment]
Addressed by #14
